### PR TITLE
backstage: add unit tests

### PR DESCRIPTION
**Pull request by burmudar**
#### This adds unit tests for:
- Queries
- SourcegraphClient

#### Other things added
* Add SearchRepoQuery, which is a variation of the normal SearchQuery but only returns repos
* It also adds the beginning of integration tests

## Test plan
Unit tests
```
pnpm test

> @sourcegraph/backstage-backend@0.1.0 test /Users/william/code/sourcegraph/client/backstage-backend
> jest --testPathIgnorePatterns src/tests/integration/

 PASS   backstage-backend  src/client/SourcegraphClient.test.ts

Test Suites: 1 skipped, 1 passed, 1 of 2 total
Tests:       2 skipped, 4 passed, 6 total
Snapshots:   0 total
Time:        1.888 s, estimated 2 s
Ran all test suites.
```

## App preview:

- [Web](https://sg-web-wb-backstage-unit-tests.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/backstage-backend/src/client/Query.ts [hunk 1]
```diff
@@ -46,7 +67,12 @@ export class SearchQuery implements Query<SearchResult[]> {
     }
 
     gql(): DocumentNode {
-        return gql(`
+        return gql(SearchQuery.raw)
```
Comment (philipp-spiess) [inline client/backstage-backend/src/client/Query.ts:70]: I think it's better if you also move the `gql` call into the `static` assignment, not sure if the syntax highlighting and code gen scripts work if you introduce indirection here.

**Review by philipp-spiess — APPROVED** [review]
Good stuff 💥

File: client/backstage-backend/src/client/Query.ts (hunk 1 above)
Comment (burmudar) [reply client/backstage-backend/src/client/Query.ts:70]: Ah good point! I'll add that thanks!

Comment (burmudar) [reply client/backstage-backend/src/client/Query.ts:70]: I've update this in the next PR. So going to merge this is in the mean time 👍🏼

**Pull request merged**